packages/flutter/lib/src/widgets/router.dart

```diff
@@ -272,7 +272,7 @@ class Router<T> extends StatefulWidget {
   ///
   /// Since this delegate is the primary consumer of the [routeInformationProvider],
   /// it must not be null if [routeInformationProvider] is not null.
-  final RouteInformationParser<T>? routeInformationParser;
+  final RouteInformationParser<Object>? routeInformationParser;
```

Comment:
Why this changing from T to Object?

Reply:
(here and elsewhere in the PR)

Reply:
@goderbauer Because I changed RouteInformationParser definition from RouteInformationParser<T> to RouteInformationParser<Object> and T is nullable so it won't be compatible and I had to change it here. 
Same goes for all changes from T to Object or Object? elsewhere